experimental/form_app/lib/src/form_widgets.dart

```diff
@@ -0,0 +1,208 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart' as intl;
+
+class FormWidgetsDemo extends StatefulWidget {
+  @override
+  _FormWidgetsDemoState createState() => _FormWidgetsDemoState();
+}
+
+class _FormWidgetsDemoState extends State<FormWidgetsDemo> {
+  final _formKey = GlobalKey<FormState>();
+  String title = "";
+  String description = "";
+  DateTime date;
+  double maxValue = 0;
+  bool brushedTeeth = false;
+  bool enableFeature = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Form widgets'),
+      ),
+      body: Form(
+        key: _formKey,
+        child: Scrollbar(
+          child: Align(
+            alignment: Alignment.topCenter,
+            child: Card(
+              child: SingleChildScrollView(
+                padding: EdgeInsets.all(16),
+                child: ConstrainedBox(
+                  constraints: BoxConstraints(maxWidth: 400),
+                  child: Column(
```

Comment:
Is it possible to present these fields using a ListView rather than a Scrollable + Column? For performance reasons, I believe the best practice is to prefer ListView (In order for Column to size itself, it has to layout all its children, which a ListView can sometimes avoid).

Reply:
I'm not sure either way is right or wrong here, since there's small number of children. In this case, doesn't this guarantee that the subtree is layed out once as opposed to potentially multiple builds / layouts with ListView?

Reply:
You're right. Now that I'm looking at this, I'm not sure. It's definitely not going to be noticeable with this few widgets, so it's only interesting from a "do we care enough to show it as the default technique every time" kind of thing.

With a Column, each child has to be given an element and a render object, laid out, and so on. With the ListView, that's done lazily, which can mean quicker first-frame time for long lists. Since those elements and render objects will be disposed once they're scrolled out of frame, though, it's possible that the Column with a small or medium amount of children would be faster at scrolling.

I may ask Goderbauer about this when we talk to him about the state restoration stuff. He knows package:widgets inside and out.